---
name: Reading Frame of a Sequence
week: 993
number: 99
points: 10
---

Authors

: [Ron K. Cytron](http://www.cs.wustl.edu/~cytron/)

Sequences of nucleotides can be read as sequences of nucleotide triplets, known as <A HREF="http://en.wikipedia.org/wiki/Codon">codons</A>.  This means there are 3 different ways of reading a nucleotide sequence.  
One starts with the first nucleotide in the sequence, one starts with the second, and one starting with the third, as shown below:

<IMG SRC="../../../extensions/frame.jpg">

Image from <a href="https://wikispaces.psu.edu/display/Biol230WFall09/Protein+Translation">https://wikispaces.psu.edu/display/Biol230WFall09/Protein+Translation</a>

* The sequences shown above are RNA.  The corresponding DNA sequences would have a `T` everywhere you see a `U`.

* Reading frame 1 begins at the first nucleotide.  Interpreting `AGT` using the <a href="http://en.wikipedia.org/wiki/DNA_codon_table">genetic code</a>,  Serine would be generated as the first amino acid of the resulting protein, if this is the proper reading frame.

* However, if the reading frame should start one base later, then the first DNA triple is `GTC`, which encodes Valine.

* Finally, if  the reading frame should start two bases later, then the first DNA triple is `TCT`, which encodes Serine.  While this protein begins with the same amino acid as in reading frame 1, the next amino acid is different, as shown in the above figure.

* There is no other reading frame of interest.  If we considered a 
reading frame that begins at the fourth nucleotide, that is the same as if
the frame began at the first nucleotide.

* **Important:** In the picture above, the reading frames are denoted 1, 2, or 3.  Keeping with computer science custom, you must compute these as 0, 1, or 2.   The unit test will not work if you fail to follow this convention!

How do we determine which reading frame is correct?  Our approach is based
on the following observations, which are somewhat simplified to suit our 
purposes.  The interested student is encouraged to take Bio 2960 for
a much more thorough and revealing treatment of this material.

* Protein translation is initiated only by a special sequence of 3 nucleotides called a  <a href="http://en.wikipedia.org/wiki/Start_codon">start codon</a>.

* Translation continues until a <a href="http://en.wikipedia.org/wiki/Stop_codon">stop codon</a> is encountered, which is also a specially designated sequence of 3 DNA nucleotides.

	Note that it is the *first* stop codon that ends the translation.  
	While the translated region could contain other start codons, which translate
	as the amino acid methionine, *any* stop codon encounered after
	the start codon ends the translation process.

* If we scan random DNA, we expect each of the four bases (A, T, C, G) to occur with probability 1 in 4.

* If we look for two specific bases, one after the other, in random DNA, scanning pairs a time, we expect the two bases to occur with probability 1 in 16 (1 in 4 times 1 in 4).

* If we scan DNA looking for a specific triplet, moving 3 bases each time we look, we expect that given triplet to occur with probability 1 in 64 (1 in 4 times 1 in 4 times 1 in 4).

	For example, we expect the start codon to occur with probability 1 in 64.

* Because there are three possible stop codons, the probability in random DNA that a given triplet is one of the three stop codons is 3 in 64, which is approximately 1 in 20.

* This means that if DNA is random, a protein sequence would be about 20 amino acids in length, once its start codon is found.

* It turns out that proteins are much longer than that, say at least 60 amino acids in length.

* If we start in the wrong reading frame, the DNA should appear to us as random, and we should obtain very short amino acid chains by translation.

Thus, the best reading frame for a sequence of DNA is the offset at which translation would produce the **longest chain of amino acids**.  The DNA sequence that produces that chain begins with the start codon.  The start codon and all subsequent codons are interpreted in groups of 3 bases.  The sequence  contains no stop codon until the end of the translated region.  In other words, the translated region may contain multiple start codons in the middle (which are translated as the amino acid methionine), but the region contains no stop codons in the middle.

Your task in this extension is to analyze a sequence of DNA to determine its best reading frame.

##Notes

You can find the Java files for this extension in the `extensions` source folder.

Locate the following files in the `biofindframe` package:

* `FindTheFrame.java` is the file you complete.

* `FindTheFrameTest.java` is a unit test for your work.  

	You will receive credit for this extension if this test (usually) passes.

* You will be working with some DNA sequences, represented in the <A HREF="http://en.wikipedia.org/wiki/FASTA_format">FASTA format</A>, which you can find in the `genomes` folder of your workspace.


##Procedure

1. The code given to you prompts the user for a genome DNA file, and then reads that file into a `char` array using methods provided by <A HREF="http://biojava.org/wiki/Main_Page">bioJava</A>.  

	If you are uncertain about `char` arrays, this would be a good time to review the relevant material in your text and in the lecture notes.

2. Your work takes place in the method `bestReadingFrame`. Open that file now and take a look at what is provided.

	There are comments in that file to direct your work, which consists of the steps described below.

	You should read these instructions and the comments *carefully* before you begin. Solutions that fail to follow the instructions will receive no credit!

3. First, define 3 `char` arrays, one for each of the possible <a href="http://en.wikipedia.org/wiki/Genetic_code#Start.2Fstop_codons">stop codons</a>: ochre, amber, and opal.

4. Next, define a `char` array named `methionine` for the start codon.

	The most common start codon is Methionine, which is encoded by the DNA sequence `AUG`.  For the purposes of this extension, Methionine is the only start codon you will consider.  

	More information about start and stop codons can be found <a href="http://en.wikipedia.org/wiki/Genetic_code#Start.2Fstop_codons">here</a>.

5. The rest of the code you write will attempt to read the DNA in each of the possible 3 reading frames. Find the best reading frame and return the index at which that best reading frame occurs.  Thus, your method will return a value as follows:

	1.  0, if reading frame 1 is the right answer.  That frame begins at index 0 of the `dna` array you are given.
	
	2.  1, if reading frame 2 is the right answer.  That frame begins at index 1 of the `dna` array you are given.

	3. 2, if reading frame 3 is the right answer.  That frame begins at index 2 of the `dna` array you are given.

	4. -1 if no reading frame exists at all

**Hint:**  Scan the DNA one base a time, looking for the longest coding sequence that begins at that base.  A coding sequence begins with a start codon and ends at the next stop codon that is found by scanning triplets.

Based on the index at which the <I>longest</I> coding sequence occurs, compute the corresponding reading frame (0, 1, or 2).  That is the value you want to return.
